src/main/src/test/java/org/geoserver/catalog/TransformFeatureTypeCallbackTest.java

```diff
@@ -0,0 +1,85 @@
+/* (c) 2023 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.catalog;
+
+import static org.geoserver.data.test.CiteTestData.PRIMITIVEGEOFEATURE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.geoserver.catalog.impl.FeatureTypeInfoImpl;
+import org.geoserver.test.GeoServerSystemTestSupport;
+import org.geotools.data.simple.SimpleFeatureSource;
+import org.junit.Before;
+import org.junit.Test;
+import org.opengis.feature.simple.SimpleFeatureType;
+
+public class TransformFeatureTypeCallbackTest extends GeoServerSystemTestSupport {
```

Comment:
Good to have, but I'd also like to see a functional test, for example, customize a property in the various DescribeFeatureTypeTest and checks it's actually producing the desired output.